csrc/executor_kernel_arg.h

```diff
@@ -109,21 +104,26 @@ class TORCH_CUDA_CU_API KernelArgumentHolder {
   //! Deserialize Kernel Argument Holder using flatbuffers
   void deserialize(const serde::KernelArgumentHolder* buffer);
 
+  // In principle, this should return a `const std::vector<void*>&`, but
+  // unfortunately cuLaunchKernel requires a non-const pointer `void**` instead
+  // of a const pointer `void* const*`
+  std::vector<void*>& getArgumentPointers(
+      kir::Kernel* kernel,
+      ExpressionEvaluator& expr_eval);
+
  private:
   std::vector<std::shared_ptr<PolymorphicValue>> arguments_;
+  std::vector<std::vector<std::byte>> kernel_argument_bytes_;
+  std::vector<void*> kernel_argument_pointers_;
```

Comment:
Are these okay to reuse?

Reply:
For most cases, yes. Indeed, we were caching the `void*` for tensor metadata before my refactor. And all the tests passes with the current change. But there is one thing that is for sure not possible to cache: RNG kernels. If we cache RNG kernels, we will be using the same seed and offset over and over again. This is why this PR is still a draft. Our current tests all pass, this indicates we should add some test checking that we are not reusing the same seed and offset...

Reply:
Makes sense, thanks!

Reply:
The missing RNG test is added in https://github.com/NVIDIA/Fuser/pull/848. Also, I am unlikely to continue working on this. Because in order to test if two tensors has exactly the same strides and shape, it takes two `memcmp` to compare these values. However, converting from the struct to byte buffer is already as cheap as two `memcpy`s, which should have similar cost as two `memcmp`s. So I don't think caching the byte buffer will provide much benefit for us.